File: frontend/src/lib/segment.ts
import type { SegmentProfileResponse } from "@/lib/api-types";

/**
 * Espelho do perfil genérico do backend. Serve de fallback enquanto o perfil real
 * carrega — mantém a tela utilizável sem piscar rótulos errados.
 */
export const GENERIC_SEGMENT_PROFILE: SegmentProfileResponse = {
  id: "generic",
  label: "Negócio",
  offering: "both",
  modules: ["clients", "services", "products", "employees"],
  terminology: {
    clients: "Clientes",
    client_singular: "cliente",
    catalog: "Produtos & Serviços",
    products: "Produtos",
    services: "Serviços",
    employees: "Funcionários",
    employee_singular: "funcionário",
    agenda: "Agenda",
    appointment_singular: "agendamento",
    transactions: "Lançamentos",
  },
  catalog_fields: {
    sku: true,
    barcode: true,
    brand: true,
    supplier: true,
    variants: true,
    inventory: true,
    duration: false,
  },
  service_examples: [],
  product_examples: [],
  catalog_categories: [],
  employee_role_examples: [],
  income_categories: ["Vendas", "Serviços"],
  expense_categories: ["Fornecedores", "Aluguel", "Salários", "Impostos", "Marketing"],
  kpis: ["total_revenue", "total_expenses", "profit", "active_clients"],
  integrations: [],
  capabilities: [],
  sells_products: true,
  sells_services: true,
};

/** Monta um placeholder "Ex.: A, B" a partir dos exemplos reais do segmento. */
export function exampleHint(examples: string[], limit = 2): string | undefined {
  if (examples.length === 0) {
    return undefined;
  }
  return `Ex.: ${examples.slice(0, limit).join(", ")}`;
}

/** Recursos do produto que só fazem sentido em certos negócios (espelha o
 * Capability do backend — vocabulário fechado, sem string solta nas telas). */
export const CAPABILITY = {
  clientRetention: "client_retention",
  inventory: "inventory",
  salesChannels: "sales_channels",
  productMargin: "product_margin",
  commissions: "commissions",
  scheduleAnalytics: "schedule_analytics",
  recurringRevenue: "recurring_revenue",
} as const;

export type CapabilityId = (typeof CAPABILITY)[keyof typeof CAPABILITY];

export function hasCapability(
  profile: { capabilities: string[] },
  capability: CapabilityId,
): boolean {
  return profile.capabilities.includes(capability);
}

File: backend/app/schemas/segment.py
from pydantic import BaseModel

from app.domain.segment.profile import SegmentProfile


class CatalogFieldPolicyResponse(BaseModel):
    sku: bool
    barcode: bool
    brand: bool
    supplier: bool
    variants: bool
    inventory: bool
    duration: bool


class SegmentTerminologyResponse(BaseModel):
    clients: str
    client_singular: str
    catalog: str
    products: str
    services: str
    employees: str
    employee_singular: str
    agenda: str
    appointment_singular: str
    transactions: str


class SegmentProfileResponse(BaseModel):
    """Contrato que o frontend usa para se adaptar ao negócio: rótulos, módulos,
    campos aplicáveis, exemplos e indicadores relevantes."""

    id: str
    label: str
    offering: str
    modules: list[str]
    terminology: SegmentTerminologyResponse
    catalog_fields: CatalogFieldPolicyResponse
    service_examples: list[str]
    product_examples: list[str]
    catalog_categories: list[str]
    income_categories: list[str]
    expense_categories: list[str]
    kpis: list[str]
    integrations: list[str]
    sells_products: bool
    sells_services: bool


def to_segment_profile_response(profile: SegmentProfile) -> SegmentProfileResponse:
    return SegmentProfileResponse(
        id=profile.id,
        label=profile.label,
        offering=profile.offering.value,
        modules=list(profile.modules),
        terminology=SegmentTerminologyResponse(
            clients=profile.terminology.clients,
            client_singular=profile.terminology.client_singular,
            catalog=profile.terminology.catalog,
            products=profile.terminology.products,
            services=profile.terminology.services,
            employees=profile.terminology.employees,
            employee_singular=profile.terminology.employee_singular,
            agenda=profile.terminology.agenda,
            appointment_singular=profile.terminology.appointment_singular,
            transactions=profile.terminology.transactions,
        ),
        catalog_fields=CatalogFieldPolicyResponse(
            sku=profile.catalog_fields.sku,
            barcode=profile.catalog_fields.barcode,
            brand=profile.catalog_fields.brand,
            supplier=profile.catalog_fields.supplier,
            variants=profile.catalog_fields.variants,
            inventory=profile.catalog_fields.inventory,
            duration=profile.catalog_fields.duration,
        ),
        service_examples=list(profile.service_examples),
        product_examples=list(profile.product_examples),
        catalog_categories=list(profile.catalog_categories),
        income_categories=list(profile.income_categories),
        expense_categories=list(profile.expense_categories),
        kpis=[metric.value for metric in profile.kpis],
        integrations=list(profile.integrations),
        sells_products=profile.sells_products,
        sells_services=profile.sells_services,
    )
